pymongo/pool.py

```diff
@@ -303,7 +316,7 @@ def __init__(self, max_pool_size=MAX_POOL_SIZE,
                  wait_queue_multiple=None, ssl_context=None,
                  ssl_match_hostname=True, socket_keepalive=True,
                  event_listeners=None, appname=None, driver=None,
-                 compression_settings=None):
+                 compression_settings=None, max_connecting=MAX_CONNECTING):
```

Comment:
This isn't configurable by users but I think it still made sense to add max_connecting to PoolOptions.